iroh-bytes/src/protocol/range_spec.rs

```diff
@@ -176,7 +176,7 @@ impl RangeSpecSeq {
         }
         let (fst_ofs, fst_val) = &self.0[0];
         let (snd_ofs, snd_val) = &self.0[1];
-        if *fst_ofs == 0 && *snd_ofs == 1 && snd_val.is_empty() {
+        if *snd_ofs - *fst_ofs == 1 && snd_val.is_empty() {
```

Comment:
🙏  for having the fortitude to review after merge.

Reply:
Take for example the one from the docs: 
`- Select ``range_a`` from blob 1234: ``[(1234, range_a), (1, empty)]``.` This will panic in debug, represent a single blob (not the first one but a single one) and won't be identified by this check

Reply:
@divagant-martian yeah, it was correct before, and I broke it with this change.